x-pack/plugins/siem/public/components/super_date_picker/index.test.tsx

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { mount } from 'enzyme';
+import * as React from 'react';
+
+import { mockGlobalState, TestProviders } from '../../mock';
+import { createStore, State } from '../../store';
+
+import { SuperDatePicker } from '.';
+
+describe('SIEM Super Date Picker', () => {
+  describe('#SuperDatePicker', () => {
+    const state: State = mockGlobalState;
+    let store = createStore(state);
+
+    beforeEach(() => {
+      store = createStore(state);
+    });
+
+    test('Can select Today (it is a relative date)', () => {
+      const wrapper = mount(
+        <TestProviders store={store}>
+          <SuperDatePicker id="global" />
+        </TestProviders>
+      );
+      wrapper
+        .find('[data-test-subj="superDatePickerToggleQuickMenuButton"]')
+        .first()
+        .simulate('click');
+      wrapper.update();
+
+      wrapper
+        .find('[data-test-subj="superDatePickerCommonlyUsed_Today"]')
+        .first()
+        .simulate('click');
+      wrapper.update();
+
+      expect(store.getState().inputs.global.timerange.kind).toBe('relative');
+      expect(store.getState().inputs.global.timerange.option).toBe('now/d');
+    });
+
+    test('Recently used date ranges', () => {
+      const wrapper = mount(
+        <TestProviders store={store}>
+          <SuperDatePicker id="global" />
+        </TestProviders>
+      );
+      wrapper
+        .find('[data-test-subj="superDatePickerToggleQuickMenuButton"]')
+        .first()
+        .simulate('click');
+      wrapper.update();
+
+      wrapper
+        .find('button.euiQuickSelect__applyButton')
+        .first()
+        .simulate('click');
+      wrapper.update();
+
+      wrapper
```

Comment:
If the previous simulated click on the `Apply` button has an intended side effect, should it be followed with a call to `expect` here?  Put differently, should we be asserting on the behavior of clicking the Apply button?  If the answer is yes, consider adding an `expect`, and then breaking up the remaining steps in this test into multiple, smaller tests. (We generally aim to have only one `expect` per test when possible, which may result in a single large test being broken up into smaller tests, with common code between tests lifted into a `beforeEach`.)

On the other hand, if clicking Apply in the previous step is necessary to setup the rest of the test, consider adding a comment that describes why we're (for example), applying a range of the `Last 15 minutes`, which appears to be the behavior when `Apply` is clicked in this case, before executing the remaining steps in this test.

Reply:
I wanted to make sure that we got multiple used date in the recently used date section, that's why we are clicking on multiple date

Reply:
As discussed, I will clean up the test to be more readable